### PR TITLE
Add CSP policy and fix errors

### PR DESCRIPTION
**Pull request by shjohnson**
### Jira link

HOTT-4465

### What?

I have added/removed/altered:

- [ ] Added CSP policy
- [ ] Fixed inline content errors and css violations
- [ ] allowed *googletagmanager violations

### Notes
There is a JQMIGRATE info message that is annoying but as i understand is only an info message and think can only be resolved by migrating to a later version but not required for this PR.

### CODE REVIEW

File: config/initializers/content_security_policy.rb [hunk 1]
```diff
@@ -4,23 +4,23 @@
...
-#   # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only
-#   # config.content_security_policy_report_only = true
-# end
+Rails.application.configure do
+  config.content_security_policy do |policy|
+    policy.default_src :self
+    policy.font_src    :self, :data
+    policy.img_src     :self, :data
+    policy.object_src  :none
+    policy.script_src  :self, 'unsafe-eval', 'unsafe-inline', '*.googletagmanager.com'
```
Comment (jebw) [inline config/initializers/content_security_policy.rb]: Strongly feel we shouldn't do this - `unsafe-eval` and `unsafe-inline` pretty much defeat any benefit of CSP

**Review by jebw — CHANGES_REQUESTED** [review]
some of those clauses are pretty self defeating

File: config/initializers/content_security_policy.rb (hunk 1 above)
Comment (willfish) [reply config/initializers/content_security_policy.rb]: We actually don't need unsafe-line or unsafe-eval (none of our sources include these - I just grepped for these)

**Review by jebw — APPROVED** [review]
LGTM :raised_hands:

**Pull request merged**